### PR TITLE
[docs] PR for Document Spring Data YB in Ecosystem (Doc8339)

### PR DESCRIPTION
**Pull request by lizayugabyte**

### CODE REVIEW

File: docs/content/latest/integrations/spring-framework/sdyb.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+## Project Dependencies
+
+The project definition includes the following dependencies:
+
+```properties
+<dependency>
+    <groupId>com.yugabyte</groupId>
+    <artifactId>spring-data-yugabytedb-ysql</artifactId>
+    <version>2.3.0</version>
```
Comment (nchandrappa) [inline docs/content/latest/integrations/spring-framework/sdyb.md:41]: Please update the version to the following, I'll reach out to update the version after GA release.

```
<version>2.3.0-RC1.b5</version>
```

File: docs/content/latest/integrations/spring-framework/sdyb.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+}
+```
+
+## Compatibility Matrix
+
+| **SDYB** | **YugabyteDB Smart Driver** | **Postgres JDBC Driver** |
+| -------- | --------------------------- | ------------------------ |
+| 2.3.0    | 42.2.7-yb-5.beta.1          | 42.2.7 or later          |
+
```
Comment (nchandrappa) [inline docs/content/latest/integrations/spring-framework/sdyb.md:141]: Please add the following line at the end 

```
To learn more about using the Spring Data YugabyteDB module, see[ Spring Data YugabyteDB example](https://github.com/yugabyte/spring-data-yugabytedb-example).
```

**Review by nchandrappa — APPROVED** [review]
LGTM. Minor review comments, please take a look. Thanks

File: docs/content/latest/integrations/spring-framework/sdyb.md (hunk 1 above)
Comment (lizayugabyte) [reply docs/content/latest/integrations/spring-framework/sdyb.md:41]: Changed

File: docs/content/latest/integrations/spring-framework/sdyb.md (hunk 2 above)
Comment (lizayugabyte) [reply docs/content/latest/integrations/spring-framework/sdyb.md:141]: Added at the end of "Examples".

**Pull request merged**